nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/sensitive/hashicorp/vault/VaultSensitivePropertyProvider.java

```diff
@@ -0,0 +1,293 @@
+import org.springframework.vault.client.VaultEndpoint;
+import org.springframework.vault.config.ClientHttpRequestFactoryFactory;
+import org.springframework.vault.core.VaultOperations;
+import org.springframework.vault.core.VaultTemplate;
+import org.springframework.vault.support.SslConfiguration;
+import org.springframework.vault.support.ClientOptions;
+
+import java.net.URI;
+import java.security.KeyStore;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Sensitive properties using Vault Transit encrypt and decrypt operations.
+ */
+public class VaultSensitivePropertyProvider implements SensitivePropertyProvider {
+    private static final Logger logger = LoggerFactory.getLogger(VaultSensitivePropertyProvider.class);
+
+    private static final String PROVIDER_NAME = "HashiCorp Vault Sensitive Property Provider";
+    static final String MATERIAL_PREFIX = "vault";
+    static final String MATERIAL_SEPARATOR = "/";
+
+    static final String VAULT_AUTH_TOKEN = "token";
+    static final String VAULT_AUTH_APP_ID = "appid";
+    static final String VAULT_AUTH_APP_ROLE = "approle";
+    static final String VAULT_AUTH_CUBBYHOLE = "cubbyhole";
+
+    private static final Set<String> VAULT_AUTH_TYPES = new HashSet<>(Arrays.asList(
+            VAULT_AUTH_TOKEN,
+            VAULT_AUTH_APP_ID,
+            VAULT_AUTH_APP_ROLE,
+            VAULT_AUTH_CUBBYHOLE
+    ));
+
+    private final VaultOperations vaultOperations;
+    private final ExternalProperties externalProperties;
+    private final String transitKeyId;
+    private final String authType;
+
+    public VaultSensitivePropertyProvider(String keyId) {
+        this(keyId, new StandardExternalPropertyLookup(null, getVaultPropertiesMapping()));
+    }
+
+    /**
+     * Constructs a {@link SensitivePropertyProvider} that uses Vault encrypt and decrypt values.
+     *
+     * @param keyId vault key spec, in the form "vault/{auth-type}/{transit-key-id}"
+     */
+    public VaultSensitivePropertyProvider(String keyId, ExternalProperties externalProperties) {
+        this.externalProperties = externalProperties;
+        transitKeyId = getTransitKey(keyId);
+        authType = getVaultAuthentication(keyId);
+
+        String serverUri = getVaultUri();
+
+        if (StringUtils.isBlank(authType) || StringUtils.isBlank(serverUri) || StringUtils.isBlank(transitKeyId))
+            throw new SensitivePropertyConfigurationException("The key cannot be empty");
+
+        VaultEndpoint vaultEndpoint = VaultEndpoint.from(URI.create(serverUri));
+        StandardVaultConfiguration config = new StandardVaultConfiguration(vaultEndpoint, externalProperties);
+
+        Resource keyStore = config.getVaultSslKeyStore();
+        Resource trustStore = config.getVaultSslTrustStore();
+        SslConfiguration sslConf = SslConfiguration.NONE;
+
+        if (keyStore != null || trustStore != null) {
```

Comment:
As far as I can see, the issue is that `SslConfiguration` requires both stores to be non-`null`.  I've re-factored the above block to be a bit more explicit + require both config types, and I've added integration tests around the behavior.

(We could drop key store completely at the moment, but I didn't want to do that in case we ever need to implement client cert authentication)